lib/data/dpa/dpClient.js

```diff
@@ -0,0 +1,280 @@
+} else {
+    client = file;
+    implName = 'file';
+}
+
+const backendId = {
+    EC_BACKEND_NULL: 0,
+    EC_BACKEND_JERASURE_RS_VAND: 1,
+    EC_BACKEND_JERASURE_RS_CAUCHY: 2,
+    EC_BACKEND_FLAT_XOR_HD: 3,
+    EC_BACKEND_ISA_L_RS_VAND: 4,
+    EC_BACKEND_SHSS: 5,
+    EC_BACKENDS_MAX: 99,
+};
+
+const checksumType = {
+    CHKSUM_NONE: 1,
+    CHKSUM_CRC32: 2,
+    CHKSUM_MD5: 3,
+    CHKSUM_TYPES_MAX: 99,
+};
+
+const ecDefault = {
+    bc_id: backendId.EC_BACKEND_JERASURE_RS_VAND, // eslint-disable-line
+    k: 9,
+    m: 1,
+    w: 8,
+    hd: 2,
+    ct: checksumType.CHKSUM_CRC32,
+};
+
+const placementPolicy = {
+    default: 'all',
+};
+
+class DpClient {
+    /**
+     * This represent our interface with the object client.
+     * @constructor
+     * @constructor
+     * @param {Object} [opts] - Contains options used by the library
+     * @param {String} [opts.pp] - placement policy
+     * @param {Object} [opts.ec] - erasure codes parameters
+     * @param {Number} [opts.ec.bc_id=0] - Backend ID
+     * @param {Number} [opts.ec.k=8] - Number of data fragments
+     * @param {Number} [opts.ec.m=4] - Number of parity fragments
+     * @param {Number} [opts.ec.w=0] - word size (in bits)
+     * @param {Number} [opts.ec.hd=0] - hamming distance (== m for Reed-Solomon)
+     * @param {Number} [opts.ec.ct=0] - checksum type
+     */
+    constructor(opts) {
+        const options = opts || {};
+        this.pp = options.pp || placementPolicy.default;
+        this.ecParams = options.ec || ecDefault;
+        this.kmin = this.ecParams.k + this.ecParams.m - this.ecParams.hd + 1;
+        this.ecParamsStr = JSON.stringify(this.ecParams);
+        this.ec = new EcLib(this.ecParams);
+        this.ec.init();
+
+        this.httpAgent = new http.Agent({ keepAlive: true });
+
+        this.setupLogging(options.log);
+    }
+
+    setupLogging(config) {
+        let options = undefined;
+        if (config !== undefined) {
+            options = {
+                level: config.logLevel,
+                dump: config.dumpLevel,
+            };
+        }
+        this.logging = new Logger('DpClient', options);
+    }
+
+    createLogger(reqUids) {
+        return reqUids ?
+            this.logging.newRequestLoggerFromSerializedUids(reqUids) :
+            this.logging.newRequestLogger();
+    }
+
+    /**
+     * This sends a PUT request to object client.
+     * @param {http.IncomingMessage} stream - Request with the data to send
+     * @param {string} stream.contentHash - hash of the data to send
+     * @param {number} size - data size in the stream
+     * @param {Object} params - parameters for key generation
+     * @param {String} params.bucketName - name of the object's bucket
+     * @param {String} params.owner - owner of the object
+     * @param {String} params.namespace - namespace of the S3 request
+     * @param {String} params.algo - algorithm for the hash
+     * @param {String} reqUids - The serialized request id
```

Comment:
the log is actually being sent through.  why not use the logger being sent rather than recreating?